src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -313,6 +318,16 @@ private static void GetGenericInfo(PEModuleSymbol moduleSymbol, TypeDefinitionHa
                 mangleName = !ReferenceEquals(_name, metadataName);
             }
 
+            // when a file-local type from source is loaded from metadata, we do a best-effort check to identify it as a file type
+            // this is needed to allow EE to bind to file types from metadata, for example.
+            if (GeneratedNameParser.TryParseFileTypeName(_name, out var displayFileName, out var ordinal, out var originalTypeName))
+            {
+                _name = originalTypeName;
+                var uncommon = GetUncommonProperties(force: true);
```

Comment:
>GetUncommonProperties(force: true); [](http://example.com/codeflow?start=31&length=35)

This looks fragile. If we already called ```GetUncommonProperties``` passing false, here we might start mutating shared instance that is supposed to be immutable.